src/main/java/com/redhat/jenkins/plugins/ci/CIBuildTrigger.java

```diff
@@ -385,6 +365,18 @@ private void startTriggerThreads() {
         return null;
     }
 
+    private static Object getLock(String name) {
+        Object lock = locks.get(name);
+        if (lock == null) {
+            Object newLock = new Object();
+            lock = locks.putIfAbsent(name, newLock);
```

Comment:
I believe this is still not correct as comparison `locks.get(name) == null` and `locks.putIfAbsent` needs to be done atomically. The whole method can be simplified to just `return locks.computeIfAbsent(name, () -> new Object())` that will be correct as well.

Reply:
Resolved in latest commit.